app/src/main/java/com/jalasoft/routesapp/ui/adminPages/userPermissions/view/PromoteUsersFragment.kt

```diff
@@ -0,0 +1,150 @@
+            override fun onQueryTextSubmit(query: String?): Boolean {
+                viewModel.searchQuery = query.toString()
+                viewModel.applyFilter()
+                return true
+            }
+
+            override fun onQueryTextChange(newText: String?): Boolean {
+                viewModel.searchQuery = newText.toString()
+                viewModel.applyFilter()
+                return true
+            }
+        })
+
+        binding.fadUserPermissions.setOnClickListener {
+            val builder = AlertDialog.Builder(binding.root.context)
+            if (userSelected.type == 0) {
+                builder.setTitle(R.string.promote_user)
+                builder.setMessage(requireContext().getString(R.string.sure_promote_user, userSelected.name))
+                builder.setPositiveButton(R.string.yes) { _, _ ->
+                    viewModel.promoteUser(userSelected)
+                }
+                builder.setNegativeButton(R.string.cancel) { _, _ ->
+                }
+                builder.show()
+            } else {
+                builder.setTitle(R.string.revoke_user)
+                builder.setMessage(requireContext().getString(R.string.sure_revoke_user, userSelected.name))
+                builder.setPositiveButton(R.string.yes) { _, _ ->
+                    viewModel.revokeUserPermission(userSelected)
+                }
+                builder.setNegativeButton(R.string.cancel) { _, _ ->
+                }
+                builder.show()
+            }
+        }
+    }
+
+    private fun observers() {
+        val errorObserver = Observer<String> { errorMessage ->
+            val builder = AlertDialog.Builder(binding.root.context)
+            builder.setTitle(errorMessage)
+            builder.setMessage(errorMessage)
+            builder.setPositiveButton(R.string.cancel) { _, _ ->
+            }
+            builder.show()
+            showProgress(false)
+        }
+        val successResult = Observer<Boolean> { successResult ->
+            if (successResult) {
+                showProgress(false)
+                Toast.makeText(context, requireContext().getString(R.string.succes_result_user), Toast.LENGTH_SHORT).show()
+                binding.fadUserPermissions.visibility = View.GONE
+                viewModel.fetchUsers()
+            } else {
+                showProgress(false)
+            }
+        }
+        viewModel.errorMessage.observe(this, errorObserver)
+        viewModel.successResult.observe(this, successResult)
+    }
+
+    private fun setRecycler() {
+        binding.recyclerUsersAdministrator.layoutManager = LinearLayoutManager(requireContext())
+        binding.recyclerUsersAdministrator.adapter = UserAdapter(mutableListOf(), this)
+    }
+
+    override fun addRevokeUserPermission(user: User) {
+        if (user.type == 0) {
+            binding.fadUserPermissions.visibility = View.VISIBLE
```

Comment:
if and else have the same value here, add the first line outside the if

Reply:
done!